view/script/trial.js

```diff
@@ -17,7 +17,37 @@
         }]
     };
     var ctx = document.getElementById('trialChart').getContext('2d');
+    var differenceValue = 0;
 
+    document.getElementById('toDate').setAttribute('min', document.getElementById('fromDate').value);
+    $('#fromDate').on('change', function changeToDateValue () {
+        var fromDateValue = document.getElementById('fromDate').value;
+        var newtoDateValue = moment(fromDateValue).add(differenceValue, 'days').format('YYYY-MM-DD');
+
+        document.getElementById('toDate').value = newtoDateValue;
+    }).on('focus', function storeDateDiff (event) {
+        var oldfromDateValue = event.target.value;
+        var oldtoDateValue = document.getElementById('toDate').value;
+
+        differenceValue = moment(oldtoDateValue).diff(moment(oldfromDateValue), 'days');
+    });
+
+    $('#check-compliance').submit(function submitAction (event) {
+        event.preventDefault();
+        $.ajax({
+            type: 'GET',
+            data: $('#check-compliance').serialize()
+        })
+          .done(function successSubmit () {
```

Comment:
I tested this and it does not update the chart.
This is correctly sends the request and gets the reply back.
But you throw the data away and reload the old data a second time.